src/main/java/hudson/plugins/sonar/action/SonarProjectActionFactory.java

```diff
@@ -113,10 +113,8 @@ private SonarProjectPageAction createProjectPage(Run<?, ?> run, List<SonarAnalys
     long endTime = run.getStartTimeInMillis() + run.getDuration();
     List<ProjectInformation> projects;
 
-    synchronized (run) {
```

Comment:
I can't see why we need to lock for this whole block, either we'll get a cached result or not.

we _may_ end up with more than one request going out if they all check around the same time, but that's better than locking up the system